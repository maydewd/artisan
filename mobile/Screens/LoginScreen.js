/**
 * Login Screen
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  AsyncStorage,
  Dimensions
} from 'react-native';
import Button from 'react-native-button'
import Icon from 'react-native-vector-icons/FontAwesome';
styles = require('../Styles/Layouts');
import {getScreenWidth, getScreenHeight, usablePercent} from '../helpers/dimension'

class LoginScreen extends Component {

    componentDidMount() {
        console.log(getScreenWidth());
        console.log(getScreenHeight());
        this.setState({username: 'devuser'});
        this.setState({password: 'securetest'});
    }

  render() {
    return (
      <View style={styles.loginScreenView}>
        <Image source = {require("../resources/Logo1.png")} style = {styles.logo} />
        <Icon.Button name="facebook" backgroundColor="#3b5998" width = {180} onPress={() => this._loginPressed()}>
            Login with Facebook
        </Icon.Button>
        <View style = {styles.groupedTextBoxes}>
          <TextInput
            style={styles.textBox}
            placeholder="Username"
            onChangeText={(username) => this.setState({username})}
          />
          <View style = {{height:5}}/>
          <TextInput
            style={styles.textBox}
            placeholder="Password"
            password = {true}
            secureTextEntry = {true}
            onChangeText={(password) => this.setState({password})}
          />
          <Button
            containerStyle={{padding:10, height:usablePercent(20), overflow:'hidden', borderRadius:4, backgroundColor: 'white'}}
            style={[styles.baseText, {fontSize: 20, color: 'pink'}]}
            onPress={() => this._submitLogin()}>
            Login
          </Button>
        </View>
        <Text style={[styles.baseText, styles.instructions]}>
          Sign Up
        </Text>
      </View>
    );
  }

  _loginPressed() {
    const { username, password } = this.state
    console.log(username);
    console.log(password);
    if (this._authenticated()) {}
      var navigator = this.props.navigator;
      navigator.replace({
          id: 'mainView'
      });
  }

  //Not functional
  _submitLogin() {
    const { username, password } = this.state

    console.log(username);
    console.log(password);
    //Username and password reach this point
    fetch("http://colab-sbx-137.oit.duke.edu:3000/api/login",
      {method: "POST",
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({username: username, password: password})})
    .then((response) => response.json())
    .then((responseData) => {
      console.log(responseData);
      if (responseData.success === true) {
        AsyncStorage.setItem('jwtToken', responseData.token, () =>
          this.props.navigator.push({id:'mainView'})
        );
      }
       return responseData;
     })
    .catch(function(err) {
      console.log("Error in Login Fetch request");
      console.log(err);
    })
    .done();
  }


  //placeholder
  _authenticated() {
    return true;
  }
}

module.exports = LoginScreen
