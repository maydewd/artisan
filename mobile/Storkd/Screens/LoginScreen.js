/**
 * Login Screen
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  TextInput
} from 'react-native';
import CenteredView from '../Screens/CenteredView'
import Button from 'react-native-button'
styles = require('../Styles/Layouts');

class LoginScreen extends Component {

  render() {
    return (
      <CenteredView>
        <Image source = {require("../resources/Logo1.png")} style = {styles.logo} />
        <Button
          containerStyle={{padding:10, height:45, overflow:'hidden', borderRadius:4, backgroundColor: 'white'}}
          style={{fontSize: 20, color: 'blue'}}>
          Facebook
        </Button>
        <View style = {styles.groupedTextBoxes}>
          <TextInput
            style={styles.textBox}
            placeholder="Username"
          />
          <TextInput
            style={styles.textBox}
            placeholder="Password"
            password = {true}
          />
          <Button
            containerStyle={{padding:10, height:45, overflow:'hidden', borderRadius:4, backgroundColor: 'white'}}
            style={{fontSize: 20, color: 'pink'}}
            onPress={() => this._loginPressed()}>
            Login
          </Button>
        </View>
        <Text style={styles.instructions}>
          Sign Up
        </Text>
      </CenteredView>
    );
  }

  _loginPressed() {
    console.log("trying to log in")
    if (this._authenticated()) {}
      var navigator = this.props.navigator;
      navigator.replace({
          id: 'mainView'
      });
  }


  //placeholder
  _authenticated() {
    return true;
  }
}

module.exports = LoginScreen
