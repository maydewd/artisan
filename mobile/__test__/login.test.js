import 'react-native';
import React from 'react';
import LoginScreen from '../Screens/LoginScreen.js';

import renderer from 'react-test-renderer';

it('Login renders correctly', () => {
  const tree = renderer.create(
    <LoginScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
