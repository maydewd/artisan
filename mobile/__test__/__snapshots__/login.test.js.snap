exports[`test Login renders correctly 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={undefined}>
  <View
    style={
      Array [
        undefined,
        Object {
          "bottom": 0,
        },
      ]
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6f2ff",
          "flexDirection": "column",
          "height": 667,
          "justifyContent": "space-around",
          "paddingTop": 40,
        }
      }>
      <Image
        source={1}
        style={
          Object {
            "height": 165.6,
            "width": 165.6,
          }
        } />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#3b5998",
                "borderRadius": 5,
              },
            ],
          ]
        }
        testID={undefined}>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 8,
              },
              Object {
                "backgroundColor": "#3b5998",
                "borderRadius": 5,
              },
              undefined,
            ]
          }
          width={180}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "FontAwesome",
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "marginRight": 10,
                },
              ]
            }>
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "fontWeight": "600",
                },
                Object {
                  "color": "white",
                },
              ]
            }>
            Login with Facebook
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "width": 337.5,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginLeft": -40,
                "padding": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#24518D",
                    "fontFamily": "Material Icons",
                    "fontSize": 25,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }>
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 16,
                "opacity": 1,
                "position": "absolute",
                "top": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#D2D2D2",
                    "fontFamily": "Arial",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "pink",
                  },
                ]
              }>
              Username
            </Text>
          </View>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            easing={[Function]}
            iconClass={[Function]}
            iconColor="#24518D"
            iconName="perm-identity"
            inputStyle={
              Object {
                "color": "pink",
              }
            }
            label="Username"
            labelStyle={
              Object {
                "color": "pink",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            returnKeyType="done"
            style={
              Array [
                Object {
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "paddingHorizontal": 16,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "pink",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined} />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "width": 337.5,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginLeft": -40,
                "padding": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#24518D",
                    "fontFamily": "Material Icons",
                    "fontSize": 25,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }>
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 16,
                "opacity": 1,
                "position": "absolute",
                "top": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#D2D2D2",
                    "fontFamily": "Arial",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "pink",
                  },
                ]
              }>
              Password
            </Text>
          </View>
          <TextInput
            easing={[Function]}
            iconClass={[Function]}
            iconColor="#24518D"
            iconName="lock"
            inputStyle={
              Object {
                "color": "pink",
              }
            }
            label="Password"
            labelStyle={
              Object {
                "color": "pink",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            password={true}
            returnKeyType="done"
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "paddingHorizontal": 16,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "pink",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined} />
        </View>
        <View
          style={
            Object {
              "height": 10,
            }
          } />
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "pink",
              "borderRadius": 4,
              "height": 40,
              "opacity": 1,
              "overflow": "hidden",
              "paddingTop": 7,
              "width": 110,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                null,
                Array [
                  Object {},
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                ],
                null,
              ]
            }>
            Login
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            },
          ]
        }>
        Sign Up
      </Text>
    </View>
  </View>
</View>
`;
